core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -5505,8 +5505,8 @@ public boolean shouldShowStackTrace() {
     /**
      * Automatically try to launch an agent when Jenkins is initialized or a new agent computer is created.
      */
-    @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "TODO needs triage")
-    public static boolean AUTOMATIC_SLAVE_LAUNCH = true;
+    @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "for script console")
+    public static boolean AUTOMATIC_AGENT_LAUNCH = SystemProperties.getBoolean(Jenkins.class.getName() + ".automaticAgentLaunch", true);
```

Comment:
It seems clear from context that this was intended to be a tunable rather than a constant. There are no references to it in the `jenkinsci` GitHub organization or in any plugins. I've made this a proper tunable using `SystemProperties` and taken the liberty to rename it to "automatic agent launch". Once this PR is merged I'll follow up with another PR to add the tunable to the documentation on `jenkins.io`.